code/modules/paperwork/paper.dm

```diff
@@ -454,6 +454,20 @@
 	ui_interact(user)
 	return ..()
 
+/// Secondary right click interaction to quickly stamp things
+/obj/item/paper/attackby_secondary(obj/item/weapon, mob/user, params)
+	var/writing_stats = weapon.get_writing_implement_details()
+
+	if(writing_stats["interaction_mode"] == MODE_STAMPING)
+		if(!user.can_read(src) || user.is_blind()) // Just leftclick instead
+			return
+		else
+			add_stamp(writing_stats["stamp_class"], rand(0,400), rand(0, 500), stamp_icon_state = writing_stats["stamp_icon_state"])
```

Comment:
400 & 500 are magic random numbers. You should get the dimensions of the paper first and replace them here.

Reply:
I took it specifically from
`add_stamp(writing_stats["stamp_class"], rand(0, 400), rand(0, 500), rand(0, 360), writing_stats["stamp_icon_state"])`

earlier in the code for the darkness / blindness check.

Reply:
I assume it's 300x400.